src/jsp/com/cubrid/plcsql/compiler/ParseTreeConverter.java

```diff
@@ -0,0 +1,1912 @@
+        assert val != null : ("invalid type name " + pcsType);
+
+        String[] split = val.split("\\.");
+        if ("Query".equals(val)) {
+            // no need to import Cursor now   TODO: remove this case later
+        } else if (val.startsWith("java.lang.") && split.length == 3) {
+            // no need to import java.lang.*
+        } else {
+            // if it is not in the java.lang package
+            addToImports(val);
+        }
+
+        return split[split.length - 1];
+    }
+
+    /*
+    private Expr
+    visitExpression(ParserRuleContext ctx) {
+        return visitExpression(ctx, null);
+    }
+     */
+
+    private Expr visitExpression(ParserRuleContext ctx, String targetType) {
+        if (ctx == null) {
+            return null;
+        }
+
+        if (targetType == null) {
+            return (Expr) visit(ctx);
+        } else {
+            Expr e = (Expr) visit(ctx);
+            if (e instanceof ExprCast) {
+                ((ExprCast) e).setTargetType(targetType);
+            }
+            return e;
+        }
+    }
+
+    private static final Map<String, String> pcsToJavaTypeMap = new TreeMap<>();
+
+    static {
+        pcsToJavaTypeMap.put("BOOLEAN", "java.lang.Boolean");
+
+        pcsToJavaTypeMap.put("CHAR", "java.lang.String");
+        pcsToJavaTypeMap.put("VARCHAR", "java.lang.String");
+        pcsToJavaTypeMap.put("STRING", "java.lang.String");
+
+        pcsToJavaTypeMap.put("NUMERIC", "java.math.BigDecimal");
+        pcsToJavaTypeMap.put("DECIMAL", "java.math.BigDecimal");
+        pcsToJavaTypeMap.put("SHORT", "java.lang.Short");
+        pcsToJavaTypeMap.put("SMALLINT", "java.lang.Short");
+        pcsToJavaTypeMap.put("INT", "java.lang.Integer");
+        pcsToJavaTypeMap.put("INTEGER", "java.lang.Integer");
+        pcsToJavaTypeMap.put("BIGINT", "java.lang.Long");
+
+        pcsToJavaTypeMap.put("FLOAT", "java.lang.Float");
+        pcsToJavaTypeMap.put("REAL", "java.lang.Float");
+        pcsToJavaTypeMap.put("DOUBLE", "java.lang.Double");
+        pcsToJavaTypeMap.put("DOUBLE PRECISION", "java.lang.Double");
+
+        pcsToJavaTypeMap.put("DATE", "java.time.LocalDate");
+        pcsToJavaTypeMap.put("TIME", "java.time.LocalTime");
+        pcsToJavaTypeMap.put("TIMESTAMP", "java.time.ZonedDateTime");
+        pcsToJavaTypeMap.put("DATETIME", "java.time.LocalDateTime");
+        pcsToJavaTypeMap.put("TIMESTAMPTZ", "java.time.ZonedDateTime");
+        pcsToJavaTypeMap.put("TIMESTAMPLTZ", "java.time.ZonedDateTime");
+        pcsToJavaTypeMap.put("DATETIMETZ", "java.time.ZonedDateTime");
+        pcsToJavaTypeMap.put("DATETIMELTZ", "java.time.ZonedDateTime");
+        pcsToJavaTypeMap.put("SET", "java.util.Set");
+        pcsToJavaTypeMap.put("MULTISET", "org.apache.commons.collections4.MultiSet");
+        pcsToJavaTypeMap.put("LIST", "java.util.List");
+        pcsToJavaTypeMap.put("SEQUENCE", "java.util.List");
+        pcsToJavaTypeMap.put("SYS_REFCURSOR", "Query");
```

Comment:
com.cubrid.plcsql.predefined.sp.SpLib.Query?

Reply:
The classes other than Query are written with full package names because package names are necessary to make a import list. Query's package com.cubrid.plcsql.predefined.sp.SpLib is always imported statically and Query is handled specially in getJavaType() method. In short, Query can be written without its package name.

Reply:
Ok. I understand.